CRM/Report/Form/Member/Detail.php

```diff
@@ -331,7 +331,11 @@ public function from() {
              LEFT JOIN civicrm_membership_payment cmp
                  ON {$this->_aliases['civicrm_membership']}.id = cmp.membership_id
              LEFT JOIN civicrm_contribution {$this->_aliases['civicrm_contribution']}
-                 ON cmp.contribution_id={$this->_aliases['civicrm_contribution']}.id\n";
+                 ON cmp.contribution_id={$this->_aliases['civicrm_contribution']}.id
+             JOIN (SELECT MAX(id) id, contact_id  
+                FROM civicrm_contribution  WHERE civicrm_contribution.financial_type_id = 2
```

Comment:
Actually - you should use the line_item table - since the data is duplicated there & one day we hope to remove the membership_payment table